package/MDAnalysis/analysis/gnm.py

```diff
@@ -154,6 +156,64 @@ def order_list(w):
     return list_map
 
 
+def generate_kirchoff(positions, bonus_groups, cutoff):
+    """Generate the Kirchhoff matrix of contacts.
+
+    This generates the neighbour matrix by generating a grid of
+    near-neighbours and then calculating which are are within
+    the cutoff.
+
+    Parameters
+    ----------
+    positions
+    bonus_groups
+    cutoff
+
+    Returns
+    -------
+    array
+            the resulting Kirchhoff matrix
+    """
+
+
+    #add the com from each bonus group to the ca_positions list
+    for item in bonus_groups:
+        #bonus = self.u.select_atoms(item)
+        positions = np.vstack((positions, item.center_of_mass()))
+
+    natoms = len(positions)
+    matrix = np.zeros((natoms, natoms), np.float64)
+
+    res_positions, grid, low_x, low_y, low_z = generate_grid(positions, cutoff)
+    cutoffsq = cutoff ** 2
+
```

Comment:
For this triple for loop, could we use `itertools.product` to save a lot of indentation